### PR TITLE
admin/build-doc: do not use system site-packages

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by tchaikov** [comment]
i think we are using the sphinx installed from pypi. see `doc-requirements.txt`. what was the issue you ran into?

**Comment by liewegas** [comment]
Maybe nothing?  Someone told me the docs needed to switch to py3 and i went down a rabbit hole of making this script run on my fc31 box.  It failed before, now it works... but maybe this isn't needed.

**Comment by liewegas** [comment]
jenkins render docs

**Comment by tchaikov** [comment]
as i think i fixed the python3 issue in #29528 already.

**Comment by tchaikov** [comment]
@liewegas i just realized that if we have Sphinx installed in system, `--system-site-packages` would
prevent `virtualenv` from installing Sphinx into the venv being prepared. and the breathe and sphinx-ditaa plugin won't be found by the system Sphinx, as they live in differnt site-packages directories in
different python installations. so we have to install Sphinx into venv, even if the system has already one installed.

could you update the commit of "admin/build-doc: use python3" with more details? as it's not related to python3. it's more related to the reason above.

File: admin/build-doc [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/sh -ex
```
Comment (tchaikov) [inline admin/build-doc]: probably we could drop "x" ?

**Comment by liewegas** [comment]
updated!

**Pull request merged**